sdk/src/main/java/com/hedera/hashgraph/sdk/Client.java

```diff
@@ -416,6 +416,7 @@ public synchronized Client setNetworkFromAddressBook(NodeAddressBook addressBook
             addressBook.nodeAddresses,
             isTransportSecurity() ? PORT_NODE_TLS : PORT_NODE_PLAIN
         ));
```

Comment:
Note: `Client::setNetworkFromAddressBook` is called from `Client::scheduleNetworkUpdate` for periodic background addressbook refresh. `Client::setNetworkFromAddressBook` currently convert the addressbook to `Map<String, AccountId>` (where string is endpoint.toString()) and passes that to `Network::setNetwork`. In other words, current behavior updates only the endpoints from the latest addressbook, not the security parameters.

If the line below which is added by this PR is only called conditionally, with the condition set to `false` during refresh, the existing behavior will be preserved. It seems to me using the latest security parameters makes sense, however I understand that there is reluctance to change the existing behavior of this code.

Reply:
I agree, it might look something like this, keeping the original interface and adding the flag:

```java
    public synchronized Client setNetworkFromAddressBook(NodeAddressBook addressBook, boolean updateAddressBook)
        throws InterruptedException, TimeoutException {
        network.setNetwork(Network.addressBookToNetwork(
            addressBook.nodeAddresses,
            isTransportSecurity() ? PORT_NODE_TLS : PORT_NODE_PLAIN
        ));
        if (updateAddressBook) {
            network.setAddressBook(addressBook);
        }
        return this;
    }

    /**
     * Replace all nodes in this Client with the nodes in the Address Book
     *
     * @param addressBook A list of nodes and their metadata
     * @return {@code this}
     * @throws InterruptedException when a thread is interrupted while it's waiting, sleeping, or otherwise occupied
     * @throws TimeoutException     when shutting down nodes
     */
    public synchronized Client setNetworkFromAddressBook(NodeAddressBook addressBook)
        throws InterruptedException, TimeoutException {
        return setNetworkFromAddressBook(addressBook, false);
    }
```